apps/prairielearn/src/pages/instructorAssessmentAccessOverrides/instructorAssessmentAccessOverrides.js

```diff
@@ -0,0 +1,149 @@
+// @ts-check
+import * as express from 'express';
+const asyncHandler = require('express-async-handler');
+import * as sqldb from '@prairielearn/postgres';
+import * as error from '@prairielearn/error';
+import { getEnrollmentForUserInCourseInstance } from '../../models/enrollment';
+import { selectUserByUid } from '../../models/user';
+
+const router = express.Router();
+const sql = sqldb.loadSqlEquiv(__filename);
+
+async function getUserIdAndCheckEnrollment({ uid, course_instance_id }) {
+  const user = await selectUserByUid(uid);
+  if (!user) {
+    throw error.make(400, `User ${uid} does not exist.`);
+  }
+
+  const enrollment = await getEnrollmentForUserInCourseInstance({
+    user_id: user.user_id,
+    course_instance_id: course_instance_id,
+  });
+  if (!enrollment) {
+    throw error.make(400, `User ${uid} is not enrolled in this course instance.`);
+  }
+
+  return user.user_id;
+}
+
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
```

Comment:
I've remembered that we need to have permission checks here. Specifically, we need to ensure that only users with "Student Data Viewer" permissions or higher can access this page, since it includes student UIDs. Furthermore, we should change the UI to reflect whether or not the user has "Course Editor" permissions, e.g. by showing a tooltip on create/edit buttons like "Only available to course content editors".

You can see an example of a page with a "Student Data Viewer" conditional check here:

https://github.com/PrairieLearn/PrairieLearn/blob/5ef40a4faa079aa5fcad56f31a0d7c77a1d0e663/apps/prairielearn/src/pages/instructorAssessmentDownloads/instructorAssessmentDownloads.js#L56-L58

Note that in this case, the tab is also conditionally hidden:

https://github.com/PrairieLearn/PrairieLearn/blob/5ef40a4faa079aa5fcad56f31a0d7c77a1d0e663/apps/prairielearn/src/pages/partials/navTabs/instructorAssessment.ejs#L6-L12

Reply:
@elisacarrillo actually, I discussed this further with Matt and we think that "Student Data Editor" would be a better permission to check for editing than "Course Editor". If you've started work on this already, this should be an easy change to make!

Reply:
Hi Nathan! Sorry I have been AFK these last two weeks, I was focusing on my midterms. To clarify which permissions I should check, is it has_course_permission_edit (check if true) or should I check if course_instance_role == 'Student Data Editor'. Also thank you for the review, you have taught me so much already and I appreciate all the help you have given me  :)

Reply:
Added has_course_permission_edit check in d485cf6

Reply:
I hope your midterms went well! `has_course_instance_permission_edit` is indeed the right thing to check.